### ISSUE TITLE
B2 SHA1 mismatch with "unverified:*" prefix

### ISSUE THREAD

**Issue by rnkn** [rclone/rclone#3654]
#### What is the problem you are having with rclone?

When running `rclone sync` on a B2 bucket, I'm getting copy failure errors where rclone thinks there's a SHA1 mismatch, but it appears to just be that B2 is prefixing the SHA1 with "unverified:"

~~~
Failed to copy: object corrupted on transfer - SHA1 mismatch (want "unverified:5dcf79685ab28048a6039253b9c81cf577bd7f07" got "5dcf79685ab28048a6039253b9c81cf577bd7f07")
~~~

As you can see, the SHA1 matches... except it doesn't...

#### What is your rclone version (output from `rclone version`)
~~~
rclone v1.49.5
- os/arch: darwin/amd64
- go version: go1.13.1
~~~

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)

macOS 10.13.6 (17G8037)
64-bit

####  Which cloud storage system are you using? (eg Google Drive)

Backblaze B2

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)

~~~
rclone --transfers 2 sync -P b2:pwr-client-uploads .
~~~

(I'm only running 2 transfers here because of a poor connection.)

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)

Ah sorry I can't give this right now as I'm in the middle of the transfer, but please let me know if it's needed.

**Comment by ncw** [comment]
How did these files get uploaded?

I found the reference to `unverified` in the docs:

https://www.backblaze.com/b2/docs/uploading.html

So it looks like these files were uploaded with the `do_not_verify` option which isn't something rclone does (rclone will use option 1 if possible and 2 if not).

You can work-around this with this flag

      --ignore-checksum    Skip post copy check of checksums.

Do you think rclone should interpret the `unverified:` prefix and use the sha1 presented?

**Comment by rnkn** [comment]
Ah wonderful... The files were uploaded by a client who is not very computer literate. Using Cyberduck, which has checksums turned off by default, because they obviously want to watch the world burn...

From rclone's perspective, given that it's verifying the checksum between B2 and rclone, not B2 and the upload point, I think it should consider this a checksum match (e.g. if the client corrupted the upload, that's kind not my problem). But this is just my opinion from this isolated case. Not sure if that might affect other cases.

Thanks for the `--ignore-checksum` hint 👍

**Comment by ncw** [comment]
> From rclone's perspective, given that it's verifying the checksum between B2 and rclone, not B2 and the upload point, I think it should consider this a checksum match (e.g. if the client corrupted the upload, that's kind not my problem).

I think that is correct.  

I've had a go at removing the `unverified:` prefix.

Can you have a go with this and see if it fixes the problem?

https://beta.rclone.org/branch/v1.49.5-245-g0f0079ff-fix-3654-b2-beta/ (uploaded in 15-30 mins)

**Comment by rnkn** [comment]
Wow thanks. Appears to be working a charm.

Perhaps instead of an error or passing silently, rclone could issue a warning that although the checksum matches, the checksum is unverified? Then leave it to the user to research further...

**Comment by ncw** [comment]
> Wow thanks. Appears to be working a charm.

Great!
 
> Perhaps instead of an error or passing silently, rclone could issue a warning that although the checksum matches, the checksum is unverified? Then leave it to the user to research further...

Unfortunately by the time the checksum gets to the layer doing the checking the fact it is unverified has been :-(

I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.50

**Comment by rnkn** [comment]
Awesome, thanks so much for your work!
